java8/src/main/java/com/dslplatform/json/runtime/ImmutableAnalyzer.java

```diff
@@ -211,30 +211,27 @@ public ImmutableDescription tryCreate(Type manifest, DslJson dslJson) {
 						}
 					}
 				}
-				if (methods.size() == ctorParams.length) {
+				if (methods.size() == paramTypes.length) {
 					writeProps = methods.values().toArray(new JsonWriter.WriteObject[0]);
 				} else {
 					return unregister(manifest, json, oldWriter, oldReader);
 				}
 				names = (fields.isEmpty() ? methods.keySet() : fields.keySet()).toArray(new String[0]);
 			}
 		}
-		final DecodePropertyInfo<JsonReader.ReadObject>[] readProps = new DecodePropertyInfo[ctorParams.length];
-		for (int i = 0; i < ctorParams.length; i++) {
-			final Type concreteType = Generics.makeConcrete(ctorParams[i].getParameterizedType(), genericMappings);
+		final DecodePropertyInfo<JsonReader.ReadObject>[] readProps = new DecodePropertyInfo[paramTypes.length];
+		for (int i = 0; i < paramTypes.length; i++) {
+			final Type concreteType = Generics.makeConcrete(paramTypes[i], genericMappings);
 			readProps[i] = new DecodePropertyInfo<>(names[i], false, false, i, false, new WriteCtor(json, concreteType, ctor));
 		}
-		final ImmutableDescription<T> converter = new ImmutableDescription<T>(
+		final ImmutableDescription<T> converter = new ImmutableDescription<>(
 				manifest,
 				defArgs,
-				new Settings.Function<Object[], T>() {
-					@Override
-					public T apply(Object[] args) {
-						try {
-							return (T) ctor.newInstance(args);
-						} catch (InstantiationException | IllegalAccessException | InvocationTargetException e) {
-							throw new RuntimeException(e);
-						}
+				args -> {

```

Comment:
I was converting lambdas to anonymous classes because Android was complaining about them. Shouldn't Android work now even without desugaring?